### ISSUE TITLE
feat(reducers): clearing firestore state on logout

### ISSUE THREAD

**Issue by RossKohler** [prescottprue/redux-firestore#114]
Is there a way to clear state.firestore.data and state.firestore.ordered? firebase.logout() seems to only clear the firebase state. Is there a similar function for clearing the Firestore state?

**Comment by RossKohler** [comment]
Just a minor correction on this, the import should be

`import { actionTypes } from 'react-redux-firestore'`

**Comment by prescottprue** [comment]
@RossKohler One more correction, I think it should actually be `redux-firestore` (`react-redux-firestore` isn't a thing to may knowledge 😆 ).

**Comment by prescottprue** [comment]
It seems like it may be due to [this line](https://github.com/prescottprue/redux-firestore/blob/master/src/reducers/orderedReducer.js#L237).

**Workaround**
Include the following in your `CLEAR_DATA` action dispatch: `meta: { collection: 'asdf' }`

It doesn't make ton of sense for now, but it should get the dispatch to clear state as expected. I'll try to get to publishing a fix soon, but is someone else gets to it, always open to PRs.

**Comment by prescottprue** [comment]
Fixed in [v0.5.8 release](https://github.com/prescottprue/redux-firestore/releases/tag/v0.5.8). The tests were also updated so that this case would be covered.

Thanks for reporting @RossKohler, reach out if it isn't working as expecting 😄

**Comment by Wgil** [comment]
Hey! Is this in the docs?

**Comment by maysam** [comment]
> ```js
> dispatch
> ```

where is the dispatch coming from here?

**Comment by maysam** [comment]
> ```js
>        dispatch({ type: actionTypes.CLEAR_DATA })
> ```

where is the dispatch coming from here?

**Comment by prescottprue** [comment]
@maysam It is supposed to come from the `onAuthStateChanged` call:

```js
onAuthStateChanged: (authData, firebase, dispatch) => {
  // Clear redux-firestore state if auth does not exist (i.e logout)
  if (!auth) {
    dispatch({ type: actionTypes.CLEAR_DATA })
  }
}
```

**Comment by nt4f04uNd** [comment]
Using 2.2.6. Such action type does not exists

**Comment by prescottprue** [comment]
@nt4f04uNd You would want to be importing this from `redux-firestore` not `react-redux-firebase` (which you listed the version of).

**Comment by MrHazimAli** [comment]
@prescottprue , how about we want to preserve everything except for one collection? is there anything like that ?

**Comment by MrHazimAli** [comment]
@prescottprue I open issue request the feature in

[https://github.com/prescottprue/redux-firestore/issues/212](https://github.com/prescottprue/redux-firestore/issues/212)

Thanks for the effort! 💯

**Comment by maysam** [comment]
thanks @MrHazimAli  ;-)

**Comment by peteruithoven** [comment]
@prescottprue In your examples you use `auth` but this doesn't exist. You probably mean `authData`, the first incomming argument?

**Comment by prescottprue** [comment]
@peteruithoven That is correct, I copied that from a place where I had named the variable auth. For clarity, here is the example fixed:

```js
onAuthStateChanged: (authData, firebase, dispatch) => {
  // Clear redux-firestore state if auth does not exist (i.e logout)
  if (!authData) {
    dispatch({ type: actionTypes.CLEAR_DATA })
  }
}
```
